app/models/katello/concerns/redhat_extensions.rb

```diff
@@ -42,10 +42,11 @@ def construct_name(family)
         end
       end
 
-      def kickstart_repos(host)
-        distros = distribution_repositories(host).where(distribution_bootable: true)
-        if distros && host&.content_facet&.content_source
-          distros.map { |distro| distro.to_hash(host.content_facet.content_source) }
+      def kickstart_repos(host, content_facet: nil)
+        distros = distribution_repositories(host, content_facet: content_facet).where(distribution_bootable: true)
```

Comment:
Do both values actually need to be passed? I'm wondering if this is a possibility to streamline the code a bit:

```
def kickstart_repos(faceted)
   facet = determine_facet(faceted)
```

```
private def determine_facet(faceted)
  case faceted
  when Host
    faceted.content_facet
   when Hostgroup
     faceted.content_facet
    ...
end
```

Reply:
If you like that better I am not averse to doing it

Reply:
Let's leave it and not make this change any larger